enumerable_methods.rb.rb

```diff
@@ -0,0 +1,100 @@
+module Enumerable
+  def my_each
+    array = self
+    array.size.times { |i| yield array[i] }
+    array
+  end
+
+  def my_each_with_index
+    array = self
+    array.length.times { |i| yield array[i], i }
+    array
+  end
+
+  def my_select
+    array = self
+    my_select_array = []
+    array.my_each { |item| my_select_array << item if yield(item) }
+    my_select_array
+  end
+
+  def my_all?
+    array = self
+    i = 0
+    all = true
+    while i < array.length
+      unless yield(array[i])
+        all = false
+        break
+      end
+      i += 1
+    end
+    all
+  end
```

Comment:
- [ ] should have working `my_all?`  identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method doesn't consider the possibility when no block or argument is given. It should return true if the enumerable contains truthy values and false if it contains at least one falsy value.

- Your method doesn't consider the possibility when a class is given as an argument `(some_enumerable.my_all?(String)).`

- Your method doesn't consider the possibility when a Regex is given as an argument `(some_enumerable.my_all?(/w/)).`

- Your method doesn't consider the possibility when a number is given as an argument `(some_enumerable.my_all?(5)).`